services/galley/schema/src/V38AddRemoteIdentifiers.hs

```diff
@@ -0,0 +1,29 @@
+module V38AddRemoteIdentifiers (addRemoteIdentifiers) where
+
+import Cassandra.Schema
+import Imports
+import Text.RawString.QQ
+
+addRemoteIdentifiers :: Migration
+addRemoteIdentifiers = Migration 38 "Add remote identifiers to conversation related tables" $ do
+  -- User table is used to answer the question: What users are part of a
+  -- conversation?  In the context of federation it is now interesting to know:
+  -- Are those users remote or local?
+  schema'
+    [r|
+      ALTER TABLE user ADD (
+        remote_user uuid,
+        remote_user_domain text
+      );
+    |]
+
+  -- The member table answers the question:  What conversations am I a member
+  -- of?  In the context of federation one now needs to know: Where is this
+  -- conversation located?
+  schema'
+    [r|
+      ALTER TABLE member ADD (
```

Comment:
So those new fields stay `null` for all local users? And the existing fields will stay `null` for all remote users? And when we look up an opaque user ID, we query this table and look at which fields are `null` to determine whether the user is local or remote?

Reply:
Similarly to the other table, maybe you want remote_user to link remote users to a local conversation?

And then user uuid is null for remote users and remote_user* is null for local users?